source/html/renderer/src/host-config.ts

```diff
@@ -133,101 +166,123 @@ function parseFactTitleDefinition(obj: any): IFactTitleDefinition {
+
+    toJSON() {
+        return {
+            maxActions: this.maxActions,
+            spacing: Enums.Spacing[this.spacing],
+            buttonSpacing: this.buttonSpacing,
+            showCard: this.showCard,
+            preExpandSingleShowCardAction: this.preExpandSingleShowCardAction,
+            actionsOrientation: Enums.Orientation[this.actionsOrientation],
+            actionAlignment: Enums.ActionAlignment[this.actionAlignment]
+        }
+    }
 }
 
-function parseActionsConfiguration(obj: any): IActionsConfig {
-    return obj ? {
+function parseActionsConfiguration(obj: any): ActionsConfig {
+    return obj ? Object.assign(new ActionsConfig(), {
         maxActions: obj["maxActions"],
         spacing: Utils.parseHostConfigEnum(Enums.Spacing, obj["spacing"], Enums.Spacing.Default),
         buttonSpacing: obj["buttonSpacing"],
         showCard: parseShowCardActionConfiguration(obj["showCard"]),
+        preExpandSingleShowCardAction: Utils.getValueOrDefault<boolean>(obj["preExpandSingleShowCardAction"], false),
         actionsOrientation: Utils.parseHostConfigEnum(Enums.Orientation, obj["actionsOrientation"], Enums.Orientation.Horizontal),
         actionAlignment: Utils.parseHostConfigEnum(Enums.ActionAlignment, obj["actionAlignment"], Enums.ActionAlignment.Left),
-    } : null;
+    }) : null;
 }
 
-export interface IHostConfig {
-    supportsInteractivity: boolean,
-    fontFamily?: string,
+export class HostConfig {
+    supportsInteractivity: boolean;
+    fontFamily?: string;
     fontSizes: {
         small: number,
         default: number,
         medium: number,
         large: number,
         extraLarge: number
-    },
+    };
     fontWeights: {
         lighter: number,
         default: number,
         bolder: number
-    },
+    };
     imageSizes: {
         small: number,
         medium: number,
         large: number
-    }
+    };
     containerStyles: {
-        default: IContainerStyleDefinition,
-        emphasis: IContainerStyleDefinition
-    },
+        default: ContainerStyleDefinition,
+        emphasis: ContainerStyleDefinition
+    };
     spacing: {
         small: number,
         default: number,
         medium: number,
         large: number,
         extraLarge: number,
         padding: number
-    },
+    };
     separator: {
         lineThickness: number,
         lineColor: string
     }
-    actions: IActionsConfig,
-    adaptiveCard: IAdaptiveCardConfig,
-    image: IImageConfig,
-    imageSet: IImageSetConfig,
-    factSet: IFactSetConfig
+    actions: ActionsConfig = new ActionsConfig();
+    adaptiveCard: AdaptiveCardConfig = new AdaptiveCardConfig();
+    image: ImageConfig = new ImageConfig();
+    imageSet: ImageSetConfig = new ImageSetConfig();
+    factSet: FactSetConfig = new FactSetConfig();
 }
 
-export function parseHostConfig(serializedConfiguration: string): IHostConfig {
+export function parseHostConfig(serializedConfiguration: string): HostConfig {
```

Comment:
Since HostConfig is now a class, this method should be moved to the class and should be renamed into simply "parse"